build/dsc/templates/dsc_type.rb.erb

```diff
@@ -4,6 +4,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
 
   provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
+    confine :operatingsystem => :windows, :powershell_version => '5.0.10514.6'
```

Comment:
You can stub out the return value of that fact on non Windows and tests
should pass again.

I haven't looked at your latest code, since I'm on phone - but if the
comparison is just a string, it might be "working" in some cases, but you
can't compare versions like that, in a general sense.
On Sep 22, 2015 8:06 PM, "James Pogran" <notifications@github.com> wrote:

> In build/dsc/templates/dsc_type.rb.erb
> <https://github.com/puppetlabs/puppetlabs-dsc/pull/99#discussion_r40167994>
> :
>
> > @@ -4,6 +4,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
> >    require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
> >
> >    provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
> > +    confine :operatingsystem => :windows, :powershell_version => '5.0.10514.6'
>
> The comparision works. I think the tests aren't creating a provider and
> thats why they are failing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-dsc/pull/99/files#r40167994>.
>